Source/Core/VideoCommon/Assets/ShaderAsset.cpp

```diff
@@ -45,26 +45,32 @@ bool ParseShaderProperties(const VideoCommon::CustomAssetLibrary::AssetID& asset
     std::transform(type.begin(), type.end(), type.begin(),
                    [](unsigned char c) { return std::tolower(c); });
 
-    if (type == "sampler2d")
-    {
-      property.m_type = ShaderProperty::Type::Type_Sampler2D;
-    }
-    else if (type == "samplercube")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerCube;
-    }
-    else if (type == "samplerarrayshared_main")
-    {
-      property.m_type = ShaderProperty::Type::Type_SamplerArrayShared_Main;
-    }
-    else if (type == "samplerarrayshared_additional")
+    static std::map<std::string, ShaderProperty::Type> str_to_type{
+        {"sampler2d", ShaderProperty::Type::Type_Sampler2D},
+        {"samplercube", ShaderProperty::Type::Type_SamplerCube},
+        {"samplerarrayshared_main", ShaderProperty::Type::Type_SamplerArrayShared_Main},
+        {"samplerarrayshared_additional", ShaderProperty::Type::Type_SamplerArrayShared_Additional},
+        {"int", ShaderProperty::Type::Type_Int},
+        {"int2", ShaderProperty::Type::Type_Int2},
+        {"int3", ShaderProperty::Type::Type_Int3},
+        {"int4", ShaderProperty::Type::Type_Int4},
+        {"float", ShaderProperty::Type::Type_Float},
+        {"float2", ShaderProperty::Type::Type_Float2},
+        {"float3", ShaderProperty::Type::Type_Float3},
+        {"float4", ShaderProperty::Type::Type_Float4},
+        {"rgb", ShaderProperty::Type::Type_RGB},
+        {"rgba", ShaderProperty::Type::Type_RGBA},
+        {"bool", ShaderProperty::Type::Type_Bool},
+    };
+
+    if (const auto result = str_to_type.find(type); result != str_to_type.end())
```

Comment:
This is small enough where an array is still almost definitely the better choice. No runtime initialization cost and much more cache friendly.

Reply:
Thanks.  I know you've mentioned it before, I just intuitively go to the map.  One of these days I'll remember all on my own :)